### ISSUE TITLE
pip generator does not collect build dependencies

### ISSUE THREAD

**Issue by theCapypara** [flatpak/flatpak-builder-tools#380]
### flatpak-builder version

1.2.3

### Linux distribution and version

Arch Linux

### Affected flatpak-builder tool

pip/flatpak-pip-generator

### Description

I was surprised to not find an issue for this yet. Please close this if I just missed one.

Currently more and more Python packages require build isolation to be built, due to them having build dependencies. This currently doesn't work properly because of the pip generator not supporting collecting these build dependencies. As such one has to manually collect and insert them.

This is seemingly blocked by https://github.com/pypa/pip/issues/7863.

#296 would also eliminate/side-step this issue and other potential headache.

This is probably the cause for at least some of these issues: #47, #90, #100, #218, #255.
Basically (almost?) any package that requires build isolation will not work because of this. Since build isolation is the recommended setup for new packages, this will become an ever-increasing issue.

**Comment by Taiko2k** [comment]
In addition to these missed build dependencies, an increasing number of Python packages themselves now have Rust dependencies. It would be nice if the Pip generator could call the Cargo generator where appropriate and add the relevant Rust build flags automatically. 

Otherwise it seems to be getting increasingly infeasible to build projects that have many Python dependencies.
